src/hyperion/experiment_plans/pin_tip_centring_plan.py

```diff
@@ -49,16 +56,34 @@ def move_pin_into_view(
         tip_x_px, tip_y_px = yield from bps.rd(oav.mxsc.pin_tip)
 
         if tip_x_px == 0:
-            LOGGER.warning(f"Pin is too long, moving -{step_size}mm")
-            yield from bps.mvr(smargon.x, -step_size)
+            min_limit = min(smargon.x.limits)
+            if float(smargon.x.user_readback.get()) - step_size_mm < min_limit:
+                LOGGER.warning(
+                    f"Pin is too long, moving -{step_size_mm} mm would cross limits, "
+                    f"moving to minimum limit {min_limit}"
+                )
+                yield from bps.mv(smargon.x, min_limit)
+                yield from bps.sleep(OAV_REFRESH_DELAY)
```

Comment:
Should: Given we're doing this sleep again below does this mean it sleeps twice? I don't think that's necessary

Reply:
no, break gets out of the for loop so we won't hit the other sleep

Reply:
Yep, sorry, my bad. I think this leads credence to the general comment of finding a slightly nicer way of doing it then. Given that we're repeating this line and the logic isn't 100% obvious